Forms/FormBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * A service which can generate a form for a given record and controller
+ */
+interface FormBuilder {
+
+	/**
+	 * @return DataObject
+	 */
+	public function getRecord();
+
+	/**
+	 * @return Controller
+	 */
+	public function getController();
```

Comment:
Note that extensions on either the record OR the controller may be invoked at each point, thus the duality of the public getters so that either can query the other component. :)